doc/Yang Model Guidelines.md

```diff
@@ -0,0 +1,253 @@
+#   YANG MODEL GUIDELINES
+
+### Rev 1.0 ###
+
+### Revision ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 1.0 | 22 Aug 2019             |      Praveen Chaudhary      | Initial version                   |
+
+
+
+## This document lists the guidelines, which will be used to write Yang Modules for Sonic. These Yang Modules will be primarily based on ABNF Schema of Sonic. 
+
+
+
+### 1. Each primary section of Config DB (i.e a dictionary in config_db.json) for Example, VLAN, VLAN_MEMBER, INTERFACE  in config DB will be mapped to a container in Yang model.
+
+Example:
+
+#### ABNF
+
+```
+"INTERFACE": {
+        ~~"Ethernet112|2a04:fxxx:40:a709::xxx/126": {
+            "scope": "global",
+            "family": "IPv6"~~
+        }
+ }
+```
+will translate to:
+
+####  Yang
+--
+```
+container INTERFACE {
+~~
+~~
+}
+```
+
+### 2. Each leaf in yang module should have same name as corresponding key in Config DB.
+
+Example:
+Leaf names are same PACKET_ACTION, IP_TYPE and PRIORITY.
+
+#### ABNF
+```
+        "NO-NSW-xxx|xxx": {
+            "PACKET_ACTION": "FORWARD",
+            "IP_TYPE": "IPv6ANY",
+            "PRIORITY": "9xxxx20",
+            .....
+            ....
+        },
+```
+
+####  Yang
+```
+            leaf PACKET_ACTION {
+                type xxxx;
+            }
+            leaf IP_TYPE {
+                type xxxx;
+            }
+            leaf PRIORITY {
+                type uint32 {
+                    range xxxx;
+                }
+            }
+```
+
+
+### 3. Hierarchy of the an objects in yang models will be same as for all the objects at same hierarchy in Redis. If any exception is created then it must be recorded properly.
```

Comment:
I would prefer to use "Data Node/Object Hierarchy " instead of "Hierarchy" ....AGREE.

I strongly believe we also must have a top level container (along with container for every list) for every model...AGREE NOW. 
Let`s add this in Guideline please. Since we want the name of the container should be same as module, please add that as well. Thanks.